spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -158,6 +178,134 @@ public Filter[] pushedFilters() {
     return pushedFilters;
   }
 
+  @Override
+  public boolean pushAggregation(Aggregation aggregation) {
+    if (!canPushDownAggregation(aggregation)) {
+      return false;
+    }
+
+    AggregateEvaluator aggregateEvaluator;
+    try {
+      List<Expression> aggregates =
+          Arrays.stream(aggregation.aggregateExpressions())
+              .map(agg -> SparkAggregates.convert(agg))
+              .collect(Collectors.toList());
+      aggregateEvaluator = AggregateEvaluator.create(schema, aggregates);
+    } catch (UnsupportedOperationException | IllegalArgumentException e) {
+      LOG.info("Skipped aggregate pushdown: " + e);
+      return false;
+    }
+
+    if (!metricsModeSupportsAggregatePushDown(aggregateEvaluator.aggregates())) {
+      return false;
+    }
+
+    TableScan scan = table.newScan();
+    ((DataTableScan) scan).setStats(true);
+    Snapshot snapshot = readSnapshot();
+    if (snapshot == null) {
+      LOG.info("Skipped aggregate pushdown: table snapshot is null");
+      return false;
+    }
+    scan = scan.useSnapshot(snapshot.snapshotId());
+    scan = configureSplitPlanning(scan);
+
+    try (CloseableIterable<FileScanTask> fileScanTasks = scan.planFiles()) {
```

Comment:
I think that Spark will also push filters first. If we don't intend to pass filters to the scan, then we will need to detect that there are filters and return `false` because we don't filter right now.

Also, it is possible to filter as long as the filter is entirely pushed down. You can check that using `ExpressionUtil.selectsPartitions`. If that returns true, then it is safe to pass the filter here!

Reply:
> I think that Spark will also push filters first. If we don't intend to pass filters to the scan, then we will need to detect that there are filters and return false because we don't filter right now.

Spark actually checks if all the filters are completely pushed down before it calls pushDownAggregation. If there are postScanFilters, Spark doesn't call pushDownAggregation.

> Also, it is possible to filter as long as the filter is entirely pushed down. You can check that using ExpressionUtil.selectsPartitions. If that returns true, then it is safe to pass the filter here!

If Spark goes to the path pushDownAggregation, then all the filters must be entirely pushed down (must all be partition filters), we don't need to check in Iceberg.
However, somehow the `GenericDataFile` still contains the stats that should be filtered out by partition filter. Maybe I didn't do it correctly. I will return false for now.

Reply:
Okay, so it sounds like as long as we are not returning filters that are entirely pushed down (which @aokolnychyi fixed recently) then it should be safe to use the filters that were pushed. In that case, we still need to configure the scan with the filters that were pushed down:

```java
scan.filter(filterExpression())
```

Reply:
Okay, I don't think this is something we need to fix right now. I see that if there are filters, we don't push the aggregation. That is fine for now, especially since I don't see the fix I thought was in that avoids returning filters that are completely handled by Iceberg.

Reply:
Added this. Thanks